index.bs

```diff
@@ -3844,19 +3850,36 @@ The [=remote end steps=] with |command parameters| are:
    1. Otherwise, set the |context|'s [=layout viewport=] to the
       implementation-defined default.
 
+1. Run the [[cssom-view-1#resizing-viewports]] steps.
+
 1. If |command parameters| [=map/contains=] the <code>devicePixelRatio</code> field:
 
    1. Let |device pixel ratio| be the |command
       parameters|["<code>devicePixelRatio</code>"].
 
-   1. If |device pixel ratio| is not null, change the size of the [=CSS Pixel=]
-      in |context|'s [=layout viewport=] such that it corresponds to |device
-      pixel ratio| in device pixels.
+   1. For the |context| and all descendant traversables:
 
-   1. Otherwise, set the size of the [=CSS Pixel=] in |context|'s [=layout
-      viewport=] to the implementation-defined default.
+      1. If |device pixel ratio| is not null:
```

Comment:
I think we also need to cover the case of:

1) window.devicePixelRatio
2) screenshots being generated using the emulated devicePixelRatio

Reply:
I've added the step for `window.devicePixelRatio` with patching `determine the device pixel ratio` algorithm, and I think it should cover capturing a screenshot, since in [the render document to a canvas algorithm](https://w3c.github.io/webdriver-bidi/#render-document-to-a-canvas) we retrieve `ratio` from `determine the device pixel ratio` (point 1).